design/README.md

```diff
@@ -0,0 +1,103 @@
+# aws-route53-cloudfront module
+<!--BEGIN STABILITY BANNER-->
+
+---
+
+![Stability: Experimental](https://img.shields.io/badge/stability-Experimental-important.svg?style=for-the-badge)
+
+> All classes are under active development and subject to non-backward compatible changes or removal in any
+> future version. These are not subject to the [Semantic Versioning](https://semver.org/) model.
+> This means that while you may use them, you may need to update your source code when upgrading to a newer version of this package.
+
+---
+<!--END STABILITY BANNER-->
+
+| **Reference Documentation**:| <span style="font-weight: normal">https://docs.aws.amazon.com/solutions/latest/constructs/</span>|
+|:-------------|:-------------|
+<div style="height:8px"></div>
+
+| **Language**     | **Package**        |
+|:-------------|-----------------|
+|![Python Logo](https://docs.aws.amazon.com/cdk/api/latest/img/python32.png) Python|`aws_solutions_constructs.aws_route53_cloudfront`|
+|![Typescript Logo](https://docs.aws.amazon.com/cdk/api/latest/img/typescript32.png) Typescript|`@aws-solutions-constructs/aws-route53-cloudfront`|
+|![Java Logo](https://docs.aws.amazon.com/cdk/api/latest/img/java32.png) Java|`software.amazon.awsconstructs.services.route53cloudfront`|
+
+This AWS Solutions Construct implements an AWS CloudFront distribution against a custom domain and protected with SSL certificate. It lets you specify overrides for CloudFront distribution props so that you can use this pattern with different types of origins.
+
+Here is a minimal deployable pattern definition in Typescript. It assumes that you have a domain purchased from Route53, and hence a corresponding hosted zone already existing.
+
+``` typescript
+import route53 from '@aws-cdk/aws-route53';
+import { Route53ToCloudFront } from '@aws-solutions-constructs/aws-route53-cloudfront';
+
+const hostedZone = route53.HostedZone.fromHostedZoneAttributes(this, 'HostedZone', {
+        hostedZoneId: 'Z00382491YVBQ98RPC5W9',
+        zoneName: 'example.com',
+    });
```

Comment:
Minimal deployment should have the construct create the hosted zone rather than create one outside the construct to be used as existingHostedZoneObj.